cmd/microcloud/main_init.go

```diff
@@ -651,6 +651,28 @@ func validateGatewayNet(config map[string]string, ipPrefix string, cidrValidator
 }
 
 func (c *initConfig) validateSystems(s *service.Handler) (err error) {
+	for _, sys := range c.systems {
+		if sys.MicroCephInternalNetworkSubnet == "" || sys.OVNGeneveAddr == "" {
+			continue
+		}
+
+		_, subnet, err := net.ParseCIDR(sys.MicroCephInternalNetworkSubnet)
+		if err != nil {
+			return fmt.Errorf("Failed to parse available network interface CIDR address: %q: %w", subnet, err)
+		}
+
+		underlayIP := net.ParseIP(sys.OVNGeneveAddr)
+		if underlayIP == nil {
+			return fmt.Errorf("OVN underlay IP %q is invalid", sys.OVNGeneveAddr)
+		}
+
+		if subnet.Contains(underlayIP) {
+			fmt.Printf("Warning: OVN underlay IP (%s) is shared with the Ceph cluster network (%s)\n", underlayIP.String(), subnet.String())
+
+			break
```

Comment:
Instead of breaking here should we also log a warning for other system's OVN underlay IPs that collide with the Ceph cluster network?

Reply:
I think that would get a bit noisy, and not particularly helpful. If you care about conflicting IPs with the ceph cluster network, the amount of conflicting IPs shouldn't really affect your behaviour. 

If you don't want any conflicting IPs, then one error should suffice to get you to restart the configuration. If you don't care about it, then you're probably not going to care if it's 2+ that collide versus 1.